<a href="https://colab.research.google.com/github/htapiagroup/sistemas-distribuidos-evaluacion-intersemestral-pacosapp/blob/master/MCA_SistDist_Ex1_FGO.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Equipo:<br/>
Ismael López Martínez <br/>
Ginna Monzerrat Nava Enríquez<br/>
Francisco García Ortega**

In [0]:
#importamos librerias necesarias
import pandas as pd, time
import psutil, os
import numpy as np
from numpy import genfromtxt
import psutil, os

**1.  Adquirir los datos** 

In [2]:
!wget https://archive.org/download/nycTaxiTripData2013/trip_data.7z

--2019-06-19 00:52:36--  https://archive.org/download/nycTaxiTripData2013/trip_data.7z
Resolving archive.org (archive.org)... 207.241.224.2
Connecting to archive.org (archive.org)|207.241.224.2|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://ia902501.us.archive.org/1/items/nycTaxiTripData2013/trip_data.7z [following]
--2019-06-19 00:52:36--  https://ia902501.us.archive.org/1/items/nycTaxiTripData2013/trip_data.7z
Resolving ia902501.us.archive.org (ia902501.us.archive.org)... 207.241.228.211
Connecting to ia902501.us.archive.org (ia902501.us.archive.org)|207.241.228.211|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4102781969 (3.8G) [application/x-7z-compressed]
Saving to: ‘trip_data.7z’

trip_data.7z        100%[===================>]   3.82G  11.3MB/s    in 5m 54s  

2019-06-19 00:58:30 (11.1 MB/s) - ‘trip_data.7z’ saved [4102781969/4102781969]



In [3]:
#contenido del del archivo de datos 
!7z l /content/trip_data.7z


7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.UTF-8,Utf16=on,HugeFiles=on,64 bits,2 CPUs Intel(R) Xeon(R) CPU @ 2.30GHz (306F0),ASM,AES-NI)

Scanning the drive for archives:
  0M Scan /content/                   1 file, 4102781969 bytes (3913 MiB)

Listing archive: /content/trip_data.7z

--
Path = /content/trip_data.7z
Type = 7z
Physical Size = 4102781969
Headers Size = 433
Method = LZMA:24
Solid = -
Blocks = 12

   Date      Time    Attr         Size   Compressed  Name
------------------- ----- ------------ ------------  ------------------------
2014-05-12 22:02:53 ....A   2459600863    345687208  trip_data_1.csv
2014-01-15 06:55:32 ....A   2502278613    358624764  trip_data_10.csv
2014-01-15 07:34:24 ....A   2395449056    342987617  trip_data_11.csv
2014-01-15 08:09:28 ....A   2327237611    331056148  trip_data_12.csv
2014-05-12 22:08:09 ....A   2328673265    327374725  trip_data_2.csv
2013

**2. Se descomprime el primer archivo correspondiente al mes de enero **

In [4]:
#se descomprimen los primeros 4 archivos de datos correspondiente al mes de enero
!7z e /content/trip_data.7z -o/content/ trip_data_1.csv



7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.UTF-8,Utf16=on,HugeFiles=on,64 bits,2 CPUs Intel(R) Xeon(R) CPU @ 2.30GHz (306F0),ASM,AES-NI)

Scanning the drive for archives:
  0M Scan /content/                   1 file, 4102781969 bytes (3913 MiB)

Extracting archive: /content/trip_data.7z
--
Path = /content/trip_data.7z
Type = 7z
Physical Size = 4102781969
Headers Size = 433
Method = LZMA:24
Solid = -
Blocks = 12

  0%      0% - trip_data_1.csv                        1% - trip_data_1.csv                        2% - trip_data_1.csv                        3% - trip_data_1.csv                        4% - trip_data_1.csv                        5% - trip_data_1.csv

In [13]:
ls

sample_data/  trip_data_1.csv*  trip_data.7z


**3. Responder las siguientes preguntas:**
A. Cuantas columnas contiene cada archivo de datos descomprimido
B. Cuantos renglones tiene cada archivo

A continuacion se muestra la cantidad de columnas y renglones que contiene el archivo:

In [22]:
df = pd.read_csv('trip_data_1.csv')
df.shape

(14776615, 14)

**14776615**  renglones y **14** columnas

**4. Identificar si existen renglones con errores en los datos, por ejemplo si hay columnas de mas (o de menos), si hay campos vacios, etc. Si se detectan renglones con errores:**

**A. indicar claramente cuantos son los renglones con errores y a que archivos corresponden**

4.1 deteccion de campos vacios en el dataframe

In [23]:
df.isnull().sum()

medallion                   0
hack_license                0
vendor_id                   0
rate_code                   0
store_and_fwd_flag    7326207
pickup_datetime             0
dropoff_datetime            0
passenger_count             0
trip_time_in_secs           0
trip_distance               0
pickup_longitude            0
pickup_latitude             0
dropoff_longitude          86
dropoff_latitude           86
dtype: int64

Se encontraron 3 columnas con datos nulos, que son store_and_fwd_flag, dropoff_longitude dropoff_latitude, a continuacion se muestran los renglones afectados de cada columna

lista de renglones nulos de la columna store_and_fwd_flag 

In [24]:
df[df['store_and_fwd_flag'].isnull()]

medallion  ... dropoff_latitude
157       F6F7D02179BE915B23EF2DB57836442D  ...        40.693043
158       BE386D8524FCD16B3727DCF0A32D9B25  ...        40.768406
159       E9FF471F36A91031FE5B6D6228674089  ...        40.732376
160       E4CF46763B08F900DC762EDEB6C37D2D  ...        40.720856
161       A5D125F5550BE7822FC6EE156E37733A  ...        40.722713
162       1E65B7E2D1297CF3B2CA87888C05FE43  ...        40.743450
163       EE1513D432B07F7E0B5E2ED1EF629086  ...        40.798096
164       90D83E0D0B4FF8DE2923C2977EF22C36  ...        40.763435
165       A56F3CF07F1085B21C365C8D3CED7528  ...        40.757061
166       34191B0F711962DA5405CF75A7D520DD  ...        40.764915
167       2111CFA825ECEE9430C3F075E8DC2E5A  ...        40.740952
168       E5264BF2B41FA581408BDFD747234640  ...        40.606304
169       834DBD712775DC97F361886D81DD2D67  ...        40.824116
170       AB68F50158A41A4CCB6DFFC7F0D43F93  ...        40.741756
171       8B17A20F99817F9751EE8683E43B712E  ...        40.748432
172       69C9DCBC1090B4811286FFA548C58088  ...        40.710377
173       2F916CFD5E51E9A620F3D4F9D65D2A7B  ...        40.784355
174       942EDD26C4E3337133058C095AD23289  ...        40.850773
175       92D01E396D7AC7BD09183A7ED793E274  ...        40.751221
176       F1C63842BC47D1C30A1317560F24FD38  ...        40.710140
177       3445DDD1C9D7E53CE70ABA3EA52A71B7  ...        40.818977
178       867892F8F81A6590EB536BE1D4F6305C  ...        40.766708
179       CEBDF09A60BF166DF41F860742D4D4E1  ...        40.717068
180       E9A3CD6DAA7E1987F9779BAA22590D81  ...        40.773182
181       27326B9D714EA6D0EFAA5AF77BFFFFFC  ...        40.764259
182       282E88327FA32E0C655B1216A91125A6  ...        40.765068
183       99274EDFA48DDA7B024A6B8BCA19EAE0  ...        40.767826
184       06808F2B069EBFA72B16977B7C4672C4  ...        40.770546
185       4E9B67DAEEF658D1B17E3B00A8A2EA71  ...        40.778648
186       0E2EF8AC28760F9B39CCB8C23CE45EDD  ...        40.793419
...                                    ...  ...              ...
14591939  95B715EA4F75E724331990880E47CD4C  ...        40.738178
14591940  F1D1F07C119A0926237467B4A1EBDE6E  ...        40.818932
14591941  A87E2C633EEB9BCF863389CCE1C6D8F9  ...        40.769932
14591942  87EC61B520189EFEAD8FDFBABD756E4A  ...        40.782444
14591943  552CCF061B871F7179CC0357086CB42C  ...        40.774429
14591944  8F8780330EC9D5EF6D035D8009A778CB  ...        40.742710
14591945  73A0C178888FDAB55F59FE1A1E640373  ...        40.762466
14591946  3EC8BDDA70B148A45686D2BC098108AC  ...        40.732994
14591947  7D9F519277EFC47748CB4C3563BC9D43  ...        40.803967
14591948  75148F5ED6DECC2B4B4FAE2F61951EBB  ...        40.774624
14591949  0DC4DDFEC71A53DBB96C8557177EAD68  ...        40.714073
14591950  2043E79B290D2371D6CE1C67DC72F774  ...        40.732281
14591951  58DB8D76A5B7E2CE8E3B12B9D3E73CE2  ...        40.753235
14591952  C4D33A7E2F87EA5D50E86411BEC65E69  ...        40.705849
14591953  32078EEE852450BEB8A2E766F556C680  ...        40.780872
14591954  C81ACCB92207A4B7CD61D1B618315256  ...        40.774872
14591955  69F42CD1BA72F0F090E85F299C7B6301  ...        40.725410
14591956  B4F985AE8AF1D7EDFDC1DAD79040B83E  ...        40.714821
14591957  73F7333AB5B05308D5019662486C77C6  ...        40.667042
14591958  B6B91A6B88A368EDB9E76A8A44B3176F  ...        40.753922
14591959  B0FF182CB0F7030F0A7EA743E49BA680  ...        40.768410
14591960  ADBE863EB343E3DE5D7FEADF365C77A9  ...        40.755013
14617395  91EE4B5588AE7F49BDA7F7E1DBA8649A  ...        40.745823
14653357  B7D6B0EFACD1B8860AA51BD9C1B68329  ...         0.000000
14705491  BC7E38FFC906E7BBC8A3454AC06FC3AC  ...        40.742485
14715955  4CAC34012C8F61C4C903025CC249E4B4  ...        40.760830
14716684  BDAB74F6D244EB1E13BC0E8585AA751D  ...        40.705101
14723966  5B61670F3280083D37ED303BA4574516  ...        40.767040
14739005  91D8D99C6CD77DFC4A6F8D5736003743  ...        40.773273
14739051  7231A69E4E3D8D34664B4F7B76A2D1A3  ...        40.752575

[7

lista de renglones nulos de la columna dropoff_longitude

In [25]:
df[df['dropoff_longitude'].isnull()]

medallion  ... dropoff_latitude
2212380   94DCB9ABFB86EA427B63719CD97E4202  ...              NaN
2267795   C59C4825017C90D9AE77DF78280E34EE  ...              NaN
2300684   C89E044B41A0AF290391EF4D690B0FD1  ...              NaN
2328873   BC0CB87C5CC5B8891DD5019A826CCE69  ...              NaN
2398011   36A4360345DCAAA0938C0B268D9B5E36  ...              NaN
2474963   901E4965DF59FA006B74057ACE3F979F  ...              NaN
2603363   37BDEA2E54A3B70CBFA5B0D1C3A75FE2  ...              NaN
2618312   E245A2E55C81F0BEB86DB6C76C43CAC6  ...              NaN
2862493   CE423C54E5BC3FAED4A07B4F64742172  ...              NaN
2891958   C1BF41E042F25C272C9D9089FD311E98  ...              NaN
3107621   31569CF28D4AA3D935E60C03044A3EA5  ...              NaN
3322978   A398F5BEFFE316F9493C7722BE21CBC2  ...              NaN
3324172   4DD8918E83252001F02331E1B602B602  ...              NaN
3324217   30DAA9127DA29636A5D3CDD370FFB8C2  ...              NaN
3358843   C1BF41E042F25C272C9D9089FD311E98  ...              NaN
3397503   9676B41DDBB34450DA07E78EEBC46012  ...              NaN
3399574   E6F3F3E401F1FD04BDDDB5261CEB6D27  ...              NaN
3449964   62E9ECD44B7177AB241585042CE642A4  ...              NaN
3480864   BCAA38AFBF6EAD8AA2F0007265F94540  ...              NaN
3481491   0547B2E5F4F79E7E178CF094E2238D59  ...              NaN
3481569   F5CA165C8029F89C95379162CF7AF187  ...              NaN
3491059   25E1299A0153252AC4615AB537C3A0B3  ...              NaN
3491358   31659BE865083136053A670ABE9282F0  ...              NaN
3491678   77AABF3A899DF44D66FB2566645E4CE4  ...              NaN
3491682   793EC1873988E07304D01E753207C05F  ...              NaN
3492113   007A1FC063DA40140AF6BFB43A047444  ...              NaN
3600292   30109A4DDF2625999908DC64EB2A768D  ...              NaN
3836103   909A85052FF481603A81F6852D120D3C  ...              NaN
3892028   52488CA04A6A36C6F096F52263DAD34D  ...              NaN
3906099   F959484B059A016083C40258093E0EDC  ...              NaN
...                                    ...  ...              ...
6071653   C76C5DC2E98F73C8A3E0C63CBAAB1097  ...              NaN
6071802   5132562861C5A605C20EAB9EF86EFB37  ...              NaN
6071950   62466D7C700FF53E0F422E5A4A90B9EE  ...              NaN
6072179   2FB8CB2151849B4C0D91F7E8122F3ECB  ...              NaN
6072540   E8950C1BF2CE598E95DE0CA5C89FA4BA  ...              NaN
6076792   37FE5F082379BBA067CA7205BE2B09A9  ...              NaN
6077116   DBCD64B365502F85569B4A264B798AAE  ...              NaN
6077431   B9C36A5E217A40659BDD63CBCBD1F273  ...              NaN
6077583   FCC400DEE2F65A9DB9E178B3994A0590  ...              NaN
6085337   64207315352C1899CA48E712BDE706A6  ...              NaN
10669637  08538606A68B9A44756733917323CE4B  ...              NaN
10742602  9F4C63E44A6C97DE0EF88E537954FC33  ...              NaN
11032439  9280082BB6EC79247F47EB181181D1A4  ...              NaN
13658309  ABCB77D9D7BEE24D2B60C019A1052293  ...              NaN
13672286  E4BE52B968F62934BE62D9D4EE00DDD6  ...              NaN
13677270  54D73B60EA0189CFC8501DF46A4B7539  ...              NaN
13681512  6C8288A327BA30EC477E601902DEE83A  ...              NaN
13697570  C87EB94EBDB4DD54DAAAF08A4597C9D3  ...              NaN
13698166  B5A4D2FCD3483BBFCA9FC459C04AC97D  ...              NaN
13719030  C20A3D0C1447B05ACB4BB8C1AB7D42FE  ...              NaN
13732084  60BD4C71E31677D13B9EB22DE29E4E93  ...              NaN
13775442  FB30B64440B4A7B8DBA9903C7598AD90  ...              NaN
13779709  7757408C647BCCCCA3D310F919FFC901  ...              NaN
13806850  1AAC732E41E8D1C7E558FE8BDD21DDA3  ...              NaN
13850025  5C1944F4CB7AD7D0A7A7BC48B6C5FDB2  ...              NaN
13941571  1BF29E6BF0CD0FC4367B3C4FBC3C6209  ...              NaN
13952630  B5728C575050CA7ABF88E5617CC6178E  ...              NaN
13985582  888574548D80421EC610441C66A0F3A8  ...              NaN
14024824  A0AD2EA71AAA38E2D57900075EFA0D6F  ...              NaN
14140322  A0B6CB5FA9A245A2D0CFC71A9373E482  ...              NaN

[8

lista de renglones nulos de la columna dropoff_latitude

In [26]:
df[df['dropoff_latitude'].isnull()]

medallion  ... dropoff_latitude
2212380   94DCB9ABFB86EA427B63719CD97E4202  ...              NaN
2267795   C59C4825017C90D9AE77DF78280E34EE  ...              NaN
2300684   C89E044B41A0AF290391EF4D690B0FD1  ...              NaN
2328873   BC0CB87C5CC5B8891DD5019A826CCE69  ...              NaN
2398011   36A4360345DCAAA0938C0B268D9B5E36  ...              NaN
2474963   901E4965DF59FA006B74057ACE3F979F  ...              NaN
2603363   37BDEA2E54A3B70CBFA5B0D1C3A75FE2  ...              NaN
2618312   E245A2E55C81F0BEB86DB6C76C43CAC6  ...              NaN
2862493   CE423C54E5BC3FAED4A07B4F64742172  ...              NaN
2891958   C1BF41E042F25C272C9D9089FD311E98  ...              NaN
3107621   31569CF28D4AA3D935E60C03044A3EA5  ...              NaN
3322978   A398F5BEFFE316F9493C7722BE21CBC2  ...              NaN
3324172   4DD8918E83252001F02331E1B602B602  ...              NaN
3324217   30DAA9127DA29636A5D3CDD370FFB8C2  ...              NaN
3358843   C1BF41E042F25C272C9D9089FD311E98  ...              NaN
3397503   9676B41DDBB34450DA07E78EEBC46012  ...              NaN
3399574   E6F3F3E401F1FD04BDDDB5261CEB6D27  ...              NaN
3449964   62E9ECD44B7177AB241585042CE642A4  ...              NaN
3480864   BCAA38AFBF6EAD8AA2F0007265F94540  ...              NaN
3481491   0547B2E5F4F79E7E178CF094E2238D59  ...              NaN
3481569   F5CA165C8029F89C95379162CF7AF187  ...              NaN
3491059   25E1299A0153252AC4615AB537C3A0B3  ...              NaN
3491358   31659BE865083136053A670ABE9282F0  ...              NaN
3491678   77AABF3A899DF44D66FB2566645E4CE4  ...              NaN
3491682   793EC1873988E07304D01E753207C05F  ...              NaN
3492113   007A1FC063DA40140AF6BFB43A047444  ...              NaN
3600292   30109A4DDF2625999908DC64EB2A768D  ...              NaN
3836103   909A85052FF481603A81F6852D120D3C  ...              NaN
3892028   52488CA04A6A36C6F096F52263DAD34D  ...              NaN
3906099   F959484B059A016083C40258093E0EDC  ...              NaN
...                                    ...  ...              ...
6071653   C76C5DC2E98F73C8A3E0C63CBAAB1097  ...              NaN
6071802   5132562861C5A605C20EAB9EF86EFB37  ...              NaN
6071950   62466D7C700FF53E0F422E5A4A90B9EE  ...              NaN
6072179   2FB8CB2151849B4C0D91F7E8122F3ECB  ...              NaN
6072540   E8950C1BF2CE598E95DE0CA5C89FA4BA  ...              NaN
6076792   37FE5F082379BBA067CA7205BE2B09A9  ...              NaN
6077116   DBCD64B365502F85569B4A264B798AAE  ...              NaN
6077431   B9C36A5E217A40659BDD63CBCBD1F273  ...              NaN
6077583   FCC400DEE2F65A9DB9E178B3994A0590  ...              NaN
6085337   64207315352C1899CA48E712BDE706A6  ...              NaN
10669637  08538606A68B9A44756733917323CE4B  ...              NaN
10742602  9F4C63E44A6C97DE0EF88E537954FC33  ...              NaN
11032439  9280082BB6EC79247F47EB181181D1A4  ...              NaN
13658309  ABCB77D9D7BEE24D2B60C019A1052293  ...              NaN
13672286  E4BE52B968F62934BE62D9D4EE00DDD6  ...              NaN
13677270  54D73B60EA0189CFC8501DF46A4B7539  ...              NaN
13681512  6C8288A327BA30EC477E601902DEE83A  ...              NaN
13697570  C87EB94EBDB4DD54DAAAF08A4597C9D3  ...              NaN
13698166  B5A4D2FCD3483BBFCA9FC459C04AC97D  ...              NaN
13719030  C20A3D0C1447B05ACB4BB8C1AB7D42FE  ...              NaN
13732084  60BD4C71E31677D13B9EB22DE29E4E93  ...              NaN
13775442  FB30B64440B4A7B8DBA9903C7598AD90  ...              NaN
13779709  7757408C647BCCCCA3D310F919FFC901  ...              NaN
13806850  1AAC732E41E8D1C7E558FE8BDD21DDA3  ...              NaN
13850025  5C1944F4CB7AD7D0A7A7BC48B6C5FDB2  ...              NaN
13941571  1BF29E6BF0CD0FC4367B3C4FBC3C6209  ...              NaN
13952630  B5728C575050CA7ABF88E5617CC6178E  ...              NaN
13985582  888574548D80421EC610441C66A0F3A8  ...              NaN
14024824  A0AD2EA71AAA38E2D57900075EFA0D6F  ...              NaN
14140322  A0B6CB5FA9A245A2D0CFC71A9373E482  ...              NaN

[8

**B. crear un nuevo conjunto de datos con los errores eliminados**

procedimiento para eliminar todos los registros con valores nulos de las columnas store_and_fwd_flag, dropoff_longitude dropoff_latitude,

In [0]:
df= df.dropna(subset=["dropoff_longitude"])

In [33]:
df.isnull().sum()

medallion                   0
hack_license                0
vendor_id                   0
rate_code                   0
store_and_fwd_flag    7326121
pickup_datetime             0
dropoff_datetime            0
passenger_count             0
trip_time_in_secs           0
trip_distance               0
pickup_longitude            0
pickup_latitude             0
dropoff_longitude           0
dropoff_latitude            0
dtype: int64

In [34]:
df.shape

(14776529, 14)

Cremos un nuevo DataFrame con los valores nulos eliminados 

In [0]:
nuevoDF = pd.DataFrame()
list_ = []
list_.append(df)
nuevoDF = pd.concat(list_)


Mostramos nuevo dataFrame limpio

In [26]:
nuevoDF

medallion  ... dropoff_latitude
0         89D227B655E5C82AECF13C3F540D4CF4  ...        40.751171
1         0BD7C8F5BA12B88E0B67BED28BEA73D8  ...        40.750660
2         0BD7C8F5BA12B88E0B67BED28BEA73D8  ...        40.726002
3         DFD2202EE08F7A8DC9A57B02ACB81FE2  ...        40.759388
4         DFD2202EE08F7A8DC9A57B02ACB81FE2  ...        40.747868
5         20D9ECB2CA0767CF7A01564DF2844A3E  ...        40.743763
6         496644932DF3932605C22C7926FF0FE0  ...        40.744343
7         0B57B9633A2FECD3D3B1944AFC7471CF  ...        40.770630
8         2C0E91FF20A856C891483ED63589F982  ...        40.735336
9         2D4B95E2FA7B2E85118EC5CA4570FA58  ...        40.806370
10        E12F6AF991172EAC3553144A0AF75A19  ...        40.773815
11        E12F6AF991172EAC3553144A0AF75A19  ...        40.759052
12        78FFD9CD0CDA541F335EF8B38FB494D6  ...        40.775921
13        237F49C3ECC11F5024B254268F054384  ...        40.755325
14        3349F919AA8AE5DC9C50A3773EA45BD8  ...        40.761612
15        3349F919AA8AE5DC9C50A3773EA45BD8  ...        40.756104
16        4C005EEBAA7BF26B84B21586332488A2  ...        40.756832
17        7D99C30FCE69B1A9DD27E2AEAC9BFA0C  ...        40.776566
18        E6FBF80668FE0611AEA44FD9574A7E32  ...        40.755390
19        0C5296F3C8B16E702F8F2E06F5106552  ...        40.787575
20        D8AADA4E722C87C10E609654612630DD  ...        40.748207
21        B6FC8357E0D53B0F1FA897D536A20F5C  ...        40.732460
22        E12F6AF991172EAC3553144A0AF75A19  ...        40.763344
23        4C005EEBAA7BF26B84B21586332488A2  ...        40.777004
24        E9BA494DC1A1F26D186394EFBB8327CF  ...        40.744553
25        E6FBF80668FE0611AEA44FD9574A7E32  ...        40.679295
26        0C5296F3C8B16E702F8F2E06F5106552  ...        40.805321
27        A3B17384165197E18CA0A1BB61277EE9  ...        40.763699
28        8E189DABE265CC03FEF4BFC695B32559  ...        40.762295
29        312E0CB058D7FC1A6494EDB66D360CD2  ...        40.751053
...                                    ...  ...              ...
14776585  237F49C3ECC11F5024B254268F054384  ...        40.720100
14776586  237F49C3ECC11F5024B254268F054384  ...        40.755569
14776587  4C005EEBAA7BF26B84B21586332488A2  ...        40.729321
14776588  32D306FD3A4D7CEDCFD8D644294DD5B7  ...        40.784794
14776589  32D306FD3A4D7CEDCFD8D644294DD5B7  ...        40.759560
14776590  50296A056AF5D221536DC98ABB1E3274  ...        40.838692
14776591  C6CE309A20A523E64FDCDECF150A2EC1  ...        40.733341
14776592  C6CE309A20A523E64FDCDECF150A2EC1  ...        40.758156
14776593  5E162F2D7F569949769648C56A23C525  ...        40.617203
14776594  1836E1591EE79F810104064B945747AB  ...        40.705822
14776595  1836E1591EE79F810104064B945747AB  ...        40.760593
14776596  A5A69FAD94C3DC7854F9DDEA994B6380  ...        40.763680
14776597  5E162F2D7F569949769648C56A23C525  ...        40.753460
14776598  B36D470B76BD8F6D720A83D08891754E  ...        40.675709
14776599  47CFD616C057758044C57D1B4BEFE759  ...        40.800568
14776600  0CC08DE5645A7137059499EA11886805  ...        40.756832
14776601  0CC08DE5645A7137059499EA11886805  ...        40.760006
14776602  B7A16B1C396F4A5113257DDEB043D953  ...        40.716122
14776603  E71CDEFCB364DA0D385629612EE30642  ...        40.776237
14776604  E71CDEFCB364DA0D385629612EE30642  ...        40.777237
14776605  A8262FA0AFCB6C7229F6888EAFBDE076  ...        40.704960
14776606  A8262FA0AFCB6C7229F6888EAFBDE076  ...        40.805023
14776607  F33EF464441839C6F0DABAABBC93B45D  ...        40.776527
14776608  56CE01E7DBE0E6449FA1758F082D8884  ...        40.742931
14776609  32201027CDC62D654DC3AD9747A07C96  ...        40.770138
14776610  B33E71CD9E8FE1BE3B70FEB6E807DD15  ...        40.770672
14776611  ED160B76D5349C8AC1ECF22CD4B8D538  ...        40.724823
14776612  D83F9AC0E33F6F19869C243BE6AB6FE5  ...        40.788158
14776613  04E59442A7DDBCE515E33CD355D866E7  ...        40.770542
14776614  D30BED60331C79E3F7ACD05B325ED42F  ...        40.773636

[7

In [0]:
#definimos el método huella_de_memoria
def huella_memoria():
  '''Regresa la huella de memoria en MB usada por un proceso de Python'''
  mem = psutil.Process(os.getpid()).memory_info().rss
  return (mem / 1024**2)


**5. Crear un DataFrame usando la libreria pandas y responder lo siguiente:**
A. Indicar el tiempo en segundos, que tarda la libreria pandas en leer un archivo a un  DataFrame
B. Cual es la huella de memoria del proceso usado para generar el objeto DataFrame
C. Cuanta memoria del sistema se usa para crear el objeto anterior
D. Indicar el tiempo que tarda pandas en obtener el promedio de la distancia de viaje (trip_distance) 

In [0]:
antes = huella_memoria()
t_start = time.time()
dfnuevo = pd.read_csv('trip_data_1.csv')
t_end = time.time()
despues = huella_memoria()


A. el tiempo en segundos que tarda la libreria en leer el archivo es de : **110.60576939582825 s**

In [7]:

print('dfnuevo.read_csv(): {} s'.format(t_end-t_start)) # time [s]

dfnuevo.read_csv(): 110.60576939582825 s


B. Cual es la huella de memoria del proceso usado para generar el objeto DataFrame: **5106.921875  MB**

In [8]:
despues

5106.921875

C. memoria utilizada para crear el objeto : **4964.14453125** MB

In [9]:
print('Memoria del proceso usado para generar el dataframe: {} MB'.format(despues-antes))

Memoria del proceso usado para generar el dataframe: 4964.14453125 MB


D. Indicar el tiempo que tarda pandas en obtener el promedio de la distancia de viaje (trip_distance)

In [10]:
#agrupamos por hora para calcular el promedio
t_inicio = time.time()
resultado = dfnuevo['trip_distance'].mean()
t_fin=time.time()
print ('pandas.read_csv(): {} s'.format(t_fin-t_inicio))

pandas.read_csv(): 0.03894209861755371 s
